plugins/org.jkiss.dbeaver.ext.tidb.ui/src/org/jkiss/dbeaver/ext/tidb/ui/views/TiDBConnectionPage.java

```diff
@@ -0,0 +1,165 @@
+    private static final Log log = Log.getLog(TiDBConnectionPage.class);
+    private static final ImageDescriptor logoImage = Activator.getImageDescriptor("icons/tidb_logo.png");
+
+    private Text portText;
+    private Text hostText;
+    private Text databaseText;
+
+    @Override
+    public void createControl(Composite composite) {
+        // Composite group = new Composite(composite, SWT.NONE);
+        // group.setLayout(new GridLayout(1, true));
+        setImageDescriptor(logoImage);
+
+        ModifyListener textListener = e -> evaluateURL();
+
+        Composite addrGroup = new Composite(composite, SWT.NONE);
+        GridLayout gl = new GridLayout(1, false);
+        addrGroup.setLayout(gl);
+        GridData gd = new GridData(GridData.FILL_BOTH);
+        addrGroup.setLayoutData(gd);
+
+        {
+            Group hostGroup = UIUtils.createControlGroup(addrGroup,
+                    TiDBMessages.tidb_connection_page_label_connection, 4, GridData.FILL_HORIZONTAL, 0);
+
+            Label hostLabel = UIUtils.createControlLabel(hostGroup,
+                    TiDBMessages.tidb_connection_page_label_host);
+            hostLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            hostText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.FILL_HORIZONTAL);
+            gd.grabExcessHorizontalSpace = true;
+            hostText.setLayoutData(gd);
+            hostText.addModifyListener(textListener);
+
+            Label portLabel = UIUtils.createControlLabel(hostGroup,
+                    TiDBMessages.tidb_connection_page_label_port);
+            portLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            portText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING);
+            gd.widthHint = UIUtils.getFontHeight(portText) * 5;
+            portText.setLayoutData(gd);
+            portText.addVerifyListener(UIUtils.getIntegerVerifyListener(Locale.getDefault()));
+            portText.addModifyListener(textListener);
+
+            Label databaseLabel = UIUtils.createControlLabel(hostGroup,
+                    TiDBMessages.tidb_connection_page_label_database);
+            databaseLabel.setLayoutData(new GridData(GridData.HORIZONTAL_ALIGN_END));
+
+            databaseText = new Text(hostGroup, SWT.BORDER);
+            gd = new GridData(GridData.FILL_HORIZONTAL);
+            gd.grabExcessHorizontalSpace = true;
+            gd.horizontalSpan = 3;
+            databaseText.setLayoutData(gd);
+            databaseText.addModifyListener(textListener);
+        }
+
+        createAuthPanel(addrGroup, 1);
+        createDriverPanel(addrGroup);
+        setControl(addrGroup);
+    }
+
+    @Override
+    public boolean isComplete() {
+        return super.isComplete() && hostText != null && databaseText != null && portText != null
+                && !CommonUtils.isEmpty(hostText.getText());

```

Comment:
Can you connect to TiDB without portText.text? You have `portText.setText("");` in your code, but I suppose you can connect without the port number.

Reply:
You're right. I'll delete ` && portText != null` part.

Reply:
Am. No. I think you can add CommonUtils.isNotEmpty(portText.getText())